### PR TITLE
DP-26260 : Add field to data metadata fields so we can paste in additional content to search

### PR DESCRIPTION
**Pull request by sagar-lcm**
**Description:**
- Added configurations for "Additional content for Data Hub search" fields
- Added dependency on the Data flag field.
- Added the new field_data_hub_search_content field to the existing hook_entity_field_access to obey access like other data fields.

**Jira:** (Skip unless you are MA staff)
https://massgov.atlassian.net/browse/DP-26260

**To Test:**
- Verify the new field called "Additional content for Data Hub search" is present on Information details, Guide, Service Details,  Curated Lists, External Link, Service Page, and Binder for Collections content types.
- Verify the "Additional content for Data Hub search field" display is hidden from the content view page.
- Verify the "Additional content for Data Hub search" field visible on the add/edit form for users with the "Access data flag fields" permission (Data Administration) role.
---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)

### CODE REVIEW

File: conf/drupal/config/core.entity_form_display.node.binder.default.yml
Comment (tom-fleming) [inline conf/drupal/config/core.entity_form_display.node.binder.default.yml]: @sagar-lcm After discussing with Joe and Adam, we would like the field name to be more generic. Could we rename the field to `field_data_search_content`?

File: conf/drupal/config/field.field.node.external_data_resource.field_data_hub_search_content.yml
Comment (tom-fleming) [inline conf/drupal/config/field.field.node.external_data_resource.field_data_hub_search_content.yml]:
```suggestion
label: 'Additional content for search'
```

Comment (sagar-lcm) [reply conf/drupal/config/field.field.node.external_data_resource.field_data_hub_search_content.yml]: Fixed.

File: conf/drupal/config/core.entity_form_display.node.binder.default.yml
Comment (sagar-lcm) [reply conf/drupal/config/core.entity_form_display.node.binder.default.yml]: Fixed.

File: docroot/modules/custom/mass_fields/mass_fields.module
Comment (tom-fleming) [inline docroot/modules/custom/mass_fields/mass_fields.module]: @sagar-lcm looks like this just needs to be updated
```suggestion
      case 'field_data_search_content':
```

Comment (sagar-lcm) [reply docroot/modules/custom/mass_fields/mass_fields.module]: @tom-fleming Thank you for pointing this out, I forgot about it.

**Review by tom-fleming — APPROVED** [review]
LGTM. Waiting on review from @adamcogbill before merging

**Comment by tom-fleming** [comment]
LGTM

**Pull request merged**
